### DISCUSSION TITLE
How to bind CTRL+SHIFT+<key> in Bash?

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#4977] [Q&A] [answered]
According to discussion in [here](https://stackoverflow.com/questions/14867368/make-bash-differentiate-between-ctrl-letter-and-ctrl-shift-letter), when binding a key combo including the SHIFT key, the behavior is dependent on the terminal emulator.

So can I bind CTRL+SHIFT+n in Bash running within kitty?

First, I would need `map kitty_mod+n no_op` because this key combo is mapped to creating a new OS window and would never make it to Bash otherwise.

Next, I tried the following in Bash:

```
bind -x '"\C-N": "my_shell_func"'
```

But `my_shell_func` is triggered _without_ needing the SHIFT key despite the fact that I specified the capital `N` within the `bind` built-in.

**Comment by kovidgoyal** [comment, marked as answer]
Legacy terminal key encoding has no encoding for ctrl+shift
Convince bash to adopt
https://sw.kovidgoyal.net/kitty/keyboard-protocol/ 

or if you cant manage that, then map it to some other legacy key in
kitty.conf using
https://sw.kovidgoyal.net/kitty/faq/#how-do-i-map-key-presses-in-kitty-to-different-keys-in-the-terminal-program 

and map that in bashrc
